ARMeilleure/CodeGen/X86/Assembler.cs

```diff
@@ -1152,6 +1158,99 @@ private void WriteInstruction(Operand dest, Operand source, OperandType type, X8
             }
         }
 
+        private void WriteEvexInst(Operand dest, Operand src1, Operand src2, OperandType type, InstructionFlags flags, int opCode, bool broadcast = false, int registerWidth = 128, int maskRegisterIdx = 0, bool zeroElements = false)
+        {
+            int destIdx = dest.GetRegister().Index;
+            int src1Idx = src1.GetRegister().Index;
+            int src2Idx = src2.GetRegister().Index;
+
+            WriteByte(0x62);
+
+            // P0
+            // Extend dest register
+            bool r = (destIdx & 8) == 0;
+            // Extend src register
+            bool x = (src1Idx & 16) == 0;
+            // Extend src register
+            bool b = (src1Idx & 8) == 0;
+            // Extend dest register
+            bool rp = (destIdx & 16) == 0;
+            // Escape code index
+            byte mm = 0b00;
+
+            switch ((ushort)(opCode >> 8))
+            {
+                case 0xf00: mm = 0b01; break;
+                case 0xf38: mm = 0b10; break;
+                case 0xf3a: mm = 0b11; break;
+
+                default: Debug.Assert(false, $"Failed to EVEX encode opcode 0x{opCode:X}."); break;
+            }
+
+            WriteByte(
+                (byte)(
+                    (r ? 0x80 : 0) |
+                    (x ? 0x40 : 0) |
+                    (b ? 0x20 : 0) |
+                    (rp ? 0x10 : 0) |
+                    mm
```

Comment:
```suggestion
                    mm));
```
That's how we have been formatting the code, same for the calls below.